### PR TITLE
VIVI-13520 Add proxy url to ydl_opts if present

### PR DESCRIPTION
**Pull request by BerkeBat**
### Description

~~Pass the `PROXY_URL` environment variable into yt-dlp through `ydl_opts`, if it's present~~
Pass the `proxy_url` query param into yt-dlp through `ydl_opts`, if it's present

--------

### Checklist

Before merge:
- [x] modifications to process, processV2 and processV3 MUST be backwards compatible
(N/A - no changes)

After merge checklist:
- [ ] update commit hash of ytdl-process in vivi-box\app\package.json
- [ ] update commit hash of ytdl-process in vivi-service-ytdl\package.json

### CODE REVIEW

**Review by tristanpenman — DISMISSED** [review]

**Comment by BerkeBat** [comment]
Putting this to draft as there's some more iteration to be done

File: service.py [hunk 1]
```diff
@@ -49,7 +50,10 @@ def do_GET(self):
             'quiet': True,
             'simulate': True
         }
+
         if url.path == '/process':
+            if proxy_url and proxy_url[0] != '':
```
Comment (ethanhusband) [inline service.py:55]: Isn't `''` falsy anyway?

Comment (BerkeBat) [reply service.py:55]: Had to fiddle with this a bit; first one checks for presence of `proxy_url` (can be `list` or `None`), 2nd one checks the actual value (`proxy_url[0]`) of the param as we know it's present now

Comment (ethanhusband) [reply service.py:55]: How come we can even get an array like `['']` though? That seems like a bug

Comment (BerkeBat) [reply service.py:55]: Sanity check I suppose, make sure that the `proxy_url` query param was provided, and that it's not an empty string for whatever reason. 

My other changes (vivi-service-ytdl) don't send an empty string as the `proxy_url` value if that's what seems buggy, the implementation higher up has its own checks

**Review by deanylev — APPROVED** [review]

**Pull request merged**